protocol/communities/persistence.go

```diff
@@ -741,6 +741,33 @@ func (p *Persistence) CanceledRequestsToJoinForUser(pk string) ([]*RequestToJoin
 	return requests, nil
 }
 
+func (p *Persistence) CanceledRequestToJoinForUserForCommunityID(pk string, communityID []byte) (*RequestToJoin, error) {
+	rows, err := p.db.Query(`SELECT id,public_key,clock,ens_name,chat_id,community_id,state FROM
+	communities_requests_to_join
+	WHERE state = ? AND public_key = ? AND community_id = ?`,
+		RequestToJoinStateCanceled, pk, communityID)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	request := &RequestToJoin{}
+	for rows.Next() {
+		err = rows.Scan(&request.ID, &request.PublicKey, &request.Clock, &request.ENSName, &request.ChatID, &request.CommunityID, &request.State)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Why do you iterate over the rows if the return value is not a collection of RequestToJoins? QueryRow should be used instead. Please pay attention to: `sql.ErrNoRows` error type, in this case we shouldn't return an error to the client.